lua/vex_library/modules/server/e2helperfuncs.lua

```diff
@@ -16,19 +16,89 @@ vex.registerExtension = function(name,enabled,helpstr,...)
     E2Lib.RegisterExtension(name,enabled,helpstr,...)
 end
 
+vex.registerConstant = function(name, value, ...)
+    -- This exists because E2Lib doesn't return a passed value, which can be handy (see selfaware2).
+    E2Lib.registerConstant(string.upper(name), value)
+    return value
+end
+
+vex.newE2Table = function() return {n={},ntypes={},s={},stypes={},size=0} end
+
+local EMPTY_TABLE = vex.newE2Table()
+local istable, type = istable, type
+vex.isE2Table = function(tbl)
```

Comment:
- [x] Added `vex.isE2Table`, another utility function.
It is used in the new `vex.luaTableToE2` and existing `vex.getE2Type` functions to determine whether the given `tbl` is an E2 table.